tools/myst/host/strace.c

```diff
@@ -91,13 +91,73 @@ int myst_set_strace_filter(
     return 0;
 }
 
+int myst_set_strace_id_filter(
+    const char* id_filter,
+    myst_strace_config_t* strace_config,
+    bool tid)
+{
+    char** tokens = NULL;
+    size_t num_tokens = 0;
+    int ret = -1;
+
+    if (myst_strsplit(id_filter, ":", &tokens, &num_tokens) != 0)
+    {
+        fprintf(
+            stderr,
+            "Invalid strace-filter-tid or strace-filter-pid option '%s' "
+            "specified.\n",
+            id_filter);
+        abort();
+    }
+
+    for (size_t i = 0; i < num_tokens; ++i)
+    {
+        long id;
```

Comment:
Not sure if we should init id to some placeholder value (like -1?) here. Looks like there is a chance that this variable is not initialized, when `myst_str2long` errors out

Reply:
Good idea, added